interlocking/interlockinginterface.py

```diff
@@ -155,20 +164,43 @@ def do_two_routes_collide(self, yaramo_route_1, yaramo_route_2):
         return do_collide
 
     def free_route(self, yaramo_route, train_id: str):
-        route = self.get_route_from_yaramo_route(yaramo_route)
+        route: Route = self.get_route_from_yaramo_route(yaramo_route)
         self.track_controller.free_route(route, train_id)
         self.active_routes.remove(route)
+        route.used_by = None
 
     async def reset_route(self, yaramo_route, train_id: str):
-        route = self.get_route_from_yaramo_route(yaramo_route)
+        route: Route = self.get_route_from_yaramo_route(yaramo_route)
         self.point_controller.reset_route(route, train_id)
         self.track_controller.reset_route(route, train_id)
         self.train_detection_controller.reset_track_segments_of_route(route)
         await self.signal_controller.reset_route(route)
         self.active_routes.remove(route)
+        route.used_by = None
 
     def get_route_from_yaramo_route(self, yaramo_route):
         for route in self.routes:
             if route.yaramo_route.uuid == yaramo_route.uuid:
                 return route
         return None
+
+    def _is_route_valid_consecutive_route(self, new_route: YaramoRoute, train_id: str):
```

Comment:
Is this really a valid case? Since railway signals are only visible in one direction, iff A1->B1 is a valid route, B1->A1 cannot be a valid route, right? Or is there any kind of signal where this is possible?

Reply:
Yes, this is a constructed case. :-)

> Since railway signals are only visible in one direction,
So it is not possible to set a route from B1 to A1 if it's not in visible direction of signals?

Reply:
@dfriedenberger nevertheless I added that logic (see return statement) and another test for the value error.